### PR TITLE
feat:add course editor onboarding

### PR DESCRIPTION
**Pull request by QinyingShi**
## Summary
- Add owner-only course editor onboarding for newly created courses from manual create / lobster create entrypoints.
- Add editor onboarding steps for prompt editing, debug, settings entry, model, listen mode, price, preview, and publish.
- Harden the shared onboarding overlay for rounded highlights, portal layering, viewport edge padding, drawer targets, and stable target measurement.
- Keep create-success toast while sequencing navigation so the toast does not overlap with editor onboarding.
- Update onboarding i18n and the local execution plan with the PR2 decisions and follow-up shared-permission scope.

## Verification
- `cd src/cook-web && npm run type-check`
- `cd src/cook-web && npm run test -- src/components/shifu-edit/ShifuEdit.test.tsx src/components/onboarding/editorOnboardingSteps.test.ts`
- `cd src/cook-web && npm run test -- src/app/admin/layout.test.tsx src/components/onboarding/onboardingSteps.test.ts src/components/onboarding/editorOnboardingSteps.test.ts`
- `python scripts/check_repo_harness.py`
- `cd src/cook-web && npm run lint` (passes with existing warnings)

## Notes
- Shared-permission editor onboarding remains deferred and documented as a later iteration.
- The push hook mentioned `branch_context_manager.py pre-pr-review`, but that script is not present in this repository checkout.

## Release Notes

* **New Features**
  * Rolled out creator onboarding across admin home and the course editor with guided overlays, step targeting, and analytics.
  * Added onboarding status + completion endpoints, with per-user/per-scene persistence and idempotent completion.
  * Improved Shifu course handling by surfacing “guide course” metadata.

* **Bug Fixes**
  * Improved API base URL resolution for local development hostnames.

* **Documentation**
  * Published the “Admin Home Onboarding” documentation page.

* **Tests**
  * Added route/UI/onboarding coverage to validate behavior and edge cases.

### CODE REVIEW

File: src/api/flaskr/service/user/onboarding.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+@dataclass(frozen=True)
+class OnboardingSceneStatus:
+    completed: bool
+    completed_at: str | None
+
+
+def _serialize_datetime(value: datetime | None) -> str | None:
+    if value is None:
+        return None
+    return value.isoformat()
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/service/user/onboarding.py]: Good catch. I updated the onboarding datetime serializer to emit explicit UTC timestamps with a `Z` suffix for both UTC-naive and timezone-aware values, and added focused coverage for null, naive, and aware datetime serialization. Verified with `cd src/api && pytest tests/service/user/test_onboarding_routes.py`.

File: src/api/flaskr/service/shifu/demo_courses.py [hunk 2]
```diff
@@ -23,6 +23,52 @@ def load_demo_shifu_bids() -> Set[str]:
...
+        for candidate_title, _ in _load_shifu_demo_metadata(app, resolved_bid):
+            if candidate_title:
+                title = candidate_title
+                break
+    if not title:
+        title = (
+            "AI 师傅教学引导"
+            if resolved_language == "zh-CN"
+            else "AI-Shifu Creation Guide"
+        )
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/flaskr/service/shifu/demo_courses.py:67]: Good catch. I moved the fallback guide course title into the shared i18n JSON namespace and added a backend language-specific translation helper so the fallback is resolved from `src/i18n/` instead of Python literals. I also updated the built-in demo title filtering to read localized titles from the loaded translation map, and regenerated the frontend i18n key type file. Verified with `python scripts/check_translations.py`, focused backend pytest coverage, `npm run type-check`, focused Jest tests, and `npm run lint`.

File: src/api/tests/common/test_i18n_loader.py [hunk 3]
```diff
@@ -49,6 +49,19 @@ def test_language_fallback_to_default():
...
 
+def test_existing_language_missing_key_falls_back_to_default():
+    os.environ["SHARED_I18N_ROOT"] = str(_shared_i18n_root())
+    app = Flask(__name__)
+
+    load_translations(app)
+
+    _translations["fr-FR"].pop("module.chat.ask")
+    _translations["fr-FR"].pop("MODULE.CHAT.ASK")
+
```
[the remark this answers is not in the document]
Comment (QinyingShi) [reply src/api/tests/common/test_i18n_loader.py]: Good catch. I updated the fallback coverage to restore the mutated `fr-FR` translation entries in a `finally` block, so the shared `_translations` state is not left altered for later tests. Verified with `cd src/api && pytest tests/common/test_i18n_loader.py`.

**Comment by QinyingShi** [comment]
I checked both failing statuses after rebasing this branch onto the latest `main`.

  1. `Backend Tests / Run backend tests`
  This failure is not introduced by this PR. The failing test is:
  `tests/service/shifu/test_admin_users.py::test_grant_operator_user_referral_reward_stacks_bucket_and_expiry`

  I reproduced the same failure locally on this PR branch, and also on the current `origin/main`. In both cases the assertion fails with:
  `second_result.summary.available_credits == "1800"` vs actual `"0"`.

  So this is a baseline backend test failure on `main`, not a regression from the course editor onboarding changes in this PR.

  2. `Runtime Harness / runtime-harness`
  This failure is also not caused by the current branch contents. The job log shows it tried to pull:
  `prom/prometheus:v2.2.4`
  and failed with:
  `manifest for prom/prometheus:v2.2.4 not found`

  However, the current branch and current `origin/main` both already use:
  `prom/prometheus:v2.54.1`
  in `docker/docker-compose.dev.yml`.

  That indicates the failing run was built from a stale merge ref / stale CI state rather than the current branch contents.

  I’ve separated the backend baseline failure into a dedicated fix branch so it can be handled independently from this onboarding PR.
